pallets/nft-manager/src/lib.rs

```diff
@@ -767,6 +758,68 @@ pub mod pallet {
         // migration logic should be done in a separate function so it can be tested
         // properly.
         fn on_runtime_upgrade() -> Weight {
+            let onchain_version = Pallet::<T>::on_chain_storage_version();
+
+            log::debug!(
+                "Nft manager storage chain/current storage version: {:?} / {:?}",
+                onchain_version,
+                Pallet::<T>::current_storage_version(),
+            );
+
+            if onchain_version < 4 {
+                use frame_support::storage::unhashed;
+
+                // Owned Nfts cleanup
+                {
+                    let owned_nfts_prefix = storage::storage_prefix(b"NftManager", b"OwnedNfts");
+                    let mut key = vec![0u8; 32];
+                    key[0..32].copy_from_slice(&owned_nfts_prefix);
+                    let res = unhashed::clear_prefix(&key[0..32], None, None);
+
+                    log::info!(
+                    "✅ Cleared '{}' backend values from 'NftManager::OwnedNfts' storage prefix",
+                        res.backend
+                    );
+                    log::info!(
+                        "✅ Cleared '{}' entries from 'NftManager::OwnedNfts' storage prefix",
+                        res.unique
+                    );
+                    if res.maybe_cursor.is_some() {
+                        log::error!(
+                            "Storage prefix 'NftManager::OwnedNfts' is not completely cleared."
+                        );
+                    }
+                }
```

Comment:
Can we replace this with `<OwnedNfts<T>>::kill()`?

Reply:
The reason I opted for this was that it doesn't require keeping the storage item in the pallet code. But I can do it like that and remove the reference afterwards.